db-scheduler/src/main/java/com/github/kagkarlsson/scheduler/JdbcTaskRepository.java

```diff
@@ -108,26 +108,37 @@ public boolean createIfNotExists(Execution execution) {
     }
 
     @Override
-    public void getScheduledExecutions(Consumer<Execution> consumer) {
+    public void getScheduledExecutions(ScheduledExecutionsFilter filter, Consumer<Execution> consumer) {
         UnresolvedFilter unresolvedFilter = new UnresolvedFilter(taskResolver.getUnresolved());
         jdbcRunner.query(
-                "select * from " + tableName + " where picked = ? " + unresolvedFilter.andCondition() + " order by execution_time asc",
+                "select * from " + tableName
+                    + (filter.getIncludePicked() && unresolvedFilter.unresolved.isEmpty() ? "" : " where ")
+                    + (!filter.getIncludePicked() ? " picked = ? " : "")
+                    + unresolvedFilter.andCondition()
+                    + " order by execution_time asc",
```

Comment:
Need to refactor this, probably a bug in here..   (filter.getIncludePicked() + unresolvedFilter.andCondition())